File: task-03/Elixir/code4.exs
defmodule Diamond do
  def print_diamond(n) do
    upper_half = for i <- 0..(n-1), do: String.duplicate(" ", n - i - 1) <> String.duplicate("*", 2 * i + 1)
    lower_half = for i <- (n-2)..0, do: String.duplicate(" ", n - i - 1) <> String.duplicate("*", 2 * i + 1)
    Enum.concat(upper_half, lower_half) |> Enum.join("\n")
  end
end

n = File.read!("input.txt") |> String.trim() |> String.to_integer()
diamond = Diamond.print_diamond(n)
File.write!("output.txt", diamond)

File: task-03/Rust/code4.rs
```rust
use std::fs;
use std::io::{self, BufRead};

fn print_diamond(n: usize) -> String {
    let mut lines = Vec::new();
    for i in 0..n {
        lines.push(format!("{}{}", " ".repeat(n - i - 1), "*".repeat(2 * i + 1)));
    }
    for i in (0..n-1).rev() {
        lines.push(format!("{}{}", " ".repeat(n - i - 1), "*".repeat(2 * i + 1)));
    }
    lines.join("\n")
}

fn main() -> io::Result<()> {
    let input = fs::read_to_string("input.txt")?;
    let n: usize = input.trim().parse().expect("Invalid number");
    let diamond = print_diamond(n);
    fs::write("output.txt", diamond)?;
    Ok(())
}
```